### PR TITLE
Chore: strip unnecessary details from database errors

### PR DESCRIPTION
**Pull request by ivard**

### CODE REVIEW

File: server/keyshare/keyshareserver/server.go [hunk 1]
```diff
@@ -279,17 +278,13 @@ func (s *Server) generateResponse(ctx context.Context, user *User, authorization
...
 	}
 
-	// Indicate activity on user account
-	err := s.db.setSeen(ctx, user)
-	if err != nil {
-		s.conf.Logger.WithField("error", err).Error("Could not mark user as seen recently")
-		// Do not send to user
-	}
+	// Indicate activity on user account. Do not send error to user.
+	_ = s.db.setSeen(ctx, user)
```
Comment (bobhageman) [inline server/keyshare/keyshareserver/server.go:282]: why not just also remove `_ = `

File: server/keyshare/keyshareserver/server.go [hunk 2]
```diff
@@ -443,20 +438,14 @@ func (s *Server) verifyAuth(ctx context.Context, user *User, msg irma.KeyshareAu
...
-		s.conf.Logger.WithField("error", err).Error("Could not reset users pin check logic")
-		// Do not send to user
-	}
-	err = s.db.setSeen(ctx, user)
-	if err != nil {
-		s.conf.Logger.WithField("error", err).Error("Could not indicate user activity")
-		// Do not send to user
-	}
+	// Handle success. Do not send error to user.
+	_ = s.db.resetPinTries(ctx, user)
```
Comment (bobhageman) [inline server/keyshare/keyshareserver/server.go:442]: why not just also remove `_ = `

File: server/keyshare/keyshareserver/server.go [hunk 3]
```diff
@@ -443,20 +438,14 @@ func (s *Server) verifyAuth(ctx context.Context, user *User, msg irma.KeyshareAu
...
+
+	// Do not send error to user.
+	_ = s.db.setSeen(ctx, user)
```
Comment (bobhageman) [inline server/keyshare/keyshareserver/server.go:445]: why not just also remove `_ = `

File: server/keyshare/keyshareserver/server.go [hunk 4]
```diff
@@ -529,17 +518,13 @@ func (s *Server) updatePin(ctx context.Context, user *User, jwtt string) (irma.K
...
 	user.Secrets = UserSecrets(secrets)
 
-	// Mark pincheck as success, resetting users wait and count
-	err = s.db.resetPinTries(ctx, user)
-	if err != nil {
-		s.conf.Logger.WithField("error", err).Error("Could not reset users pin check logic")
-		// Do not send to user
-	}
+	// Mark pincheck as success, resetting users wait and count. Do not send error to user.
+	_ = s.db.resetPinTries(ctx, user)
```
Comment (bobhageman) [inline server/keyshare/keyshareserver/server.go:522]: why not just also remove `_ = `

File: server/keyshare/keyshareserver/server.go (hunk 1 above)
Comment (ivard) [reply server/keyshare/keyshareserver/server.go:282]: I'm a favour of explicitly showing that I drop a variable here. That makes it more clear in the future that we make an assumption here. If you have to add code in the future, then that might be important.

But indeed, it's just one of my habits. We would't get a warning if we remove `_ =`.

**Review by bobhageman — APPROVED** [review]

**Review by sjoerdlmkns — APPROVED** [review]

**Pull request merged**